gui/default/index.html

```diff
@@ -329,9 +332,39 @@ <h3 class="panel-title">
         </div>
       </div>
 
-      <!-- First regular row -->
+      <!-- Login form -->
+      <div ng-if="!authenticated" class="center-block">
+        <h3 translate>Authentication required</h3>
 
-      <div class="row">
+        <form ng-submit="authenticatePassword()">
+          <div class="form-group">
+            <label for="User" translate>User</label>
+            <input id="User" class="form-control" type="text" ng-model="login.username" />
+          </div>
+
+          <div class="form-group">
+            <label for="Password" translate>Password</label>
+            <input id="Password" class="form-control" type="password" ng-model="login.password" ng-trim="false" />
+          </div>
+
+          <div class="row">
+            <div class="col-md-9 login-form-messages">
+              <p ng-if="login.errors.badLogin" class="text-danger">
+                Incorrect user name or password.
+              </p>
+              <p ng-if="login.errors.failed" class="text-danger">
+                Login failed, see Syncthing logs for details.
+              </p>
+            </div>
+            <div class="col-md-3 text-right">
+              <button type="submit" class="btn btn-default" ng-disabled="login.inProgress" translate>Log in</button>
```

Comment:
```suggestion
              <button type="submit" class="btn btn-default" ng-disabled="login.inProgress || login.username === '' || login.password === ''" translate>Log in</button>
```

More of a cosmetic fix, but this disables the "log in" button when either the user name or the password is empty. Currently, you can click it always and get the error message about "incorrect user name or password" even if they are completely empty.

Reply:
Can we have a username without a password though?  Is setting the password to an empty string valid?  (Even if it means copying a hash of the empty string to the config file, if Syncthing itself doesn't support setting it.)

Reply:
Yeah, it seems you can set just the username with no password (but not the other way around), so the code should probably be only:

```
ng-disabled="login.inProgress || login.username === ''"
```